tools/Import/Importer.cs

```diff
@@ -0,0 +1,271 @@
+                });
+                Console.WriteLine($"{localLogPrefix}: Completed writes. Total={localResourceCount} secs={(int)localSw.Elapsed.TotalSeconds} speed={(int)(localResourceCount / localSw.Elapsed.TotalSeconds)} res/sec");
+            });
+            Console.WriteLine($"{globalLogPrefix}.Readers={readers}.Writers={writers}.EndPointCalls={epCalls}.Waits={waits}: total reads={resourcesRead} total writes={globalResourceCount} secs={(int)swWrites.Elapsed.TotalSeconds} read-speed={(int)(resourcesRead / swReads.Elapsed.TotalSeconds)} lines/sec write-speed={(int)(globalResourceCount / swWrites.Elapsed.TotalSeconds)} res/sec");
+        }
+
+        private static IEnumerable<string> GetLinesInBlobRange(IList<BlobItem> blobs, string logPrefix)
+        {
+            Interlocked.Increment(ref readers);
+            swReads.Start(); // just in case it was stopped by decrement logic below
+
+            var blobContainerClient = GetContainer(ConnectionString, ContainerName);
+            var lines = 0L;
+            var sw = Stopwatch.StartNew();
+            foreach (var blob in blobs)
+            {
+                if (blob.Name.EndsWith(".json", StringComparison.OrdinalIgnoreCase))
+                {
+                    continue;
+                }
+
+                using var reader = new StreamReader(blobContainerClient.GetBlobClient(blob.Name).Download().Value.Content);
+                while (!reader.EndOfStream)
+                {
+                    yield return reader.ReadLine();
+                    lines++;
+                    Interlocked.Increment(ref resourcesRead);
+                }
+            }
+
+            if (Interlocked.Decrement(ref readers) == 0)
+            {
+                swReads.Stop();
+            }
+
+            Console.WriteLine($"{logPrefix}: Completed reads. Total={lines} secs={(int)sw.Elapsed.TotalSeconds} speed={(int)(lines / sw.Elapsed.TotalSeconds)} lines/sec");
+        }
+
+        private static BlobContainerClient GetContainer(string connectionString, string containerName)
+        {
+            try
+            {
+                var blobServiceClient = new BlobServiceClient(connectionString);
+                var blobContainerClient = blobServiceClient.GetBlobContainerClient(containerName);
+
+                if (!blobContainerClient.Exists())
+                {
+                    Console.WriteLine($"Storage container {containerName} not found.");
+                    return null;
+                }
+
+                return blobContainerClient;
+            }
+            catch
+            {
+                Console.WriteLine($"Unable to parse stroage reference or connect to storage account {connectionString}.");
+                throw;
+            }
+        }
+
+        private static void PutResource(string jsonString, ref long resourceCount, ref long epCalls, ref long waits)
+        {
+            ParseJson(jsonString, out var resourceType, out var resourceId);
```

Comment:
Is there any significant performance difference between ParseJson and ParseJsonWithJObject? IndexOf may be faster than JObject parsing, but the latter ensures a valid json object and returns the id and resource type nicely.

Reply:
There is huge difference in memory allocation and, hence, perf. I did not measure precisely but gut feeling is 10x.

Reply:
in general, i'm _very_ afraid of string parsing json.  i'm not convinced we should be offering it as an option, as there are best-practice reasons against it.    JObject has evolved to be pretty light; if we really want to do better, maybe something based on Utf8JsonReader

Reply:
Any JSON reader will always loose compared to parsing of first 100 characters of the string.

Reply:
There are also forward-only jsonreaders, i.e. https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-use-dom-utf8jsonreader-utf8jsonwriter?pivots=dotnet-6-0#read-from-a-stream-using-utf8jsonreader, so you would be able to parse until you hit these two properties